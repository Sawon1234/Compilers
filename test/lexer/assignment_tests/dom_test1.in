interface List<E> {
fun elements() : Iterable<E>;
}
class Nil() extends List<Nothing> {
fun elements() : Iterable<Nothing>
{ return []; }
}
class Cons<E>(E head, List<E> tail)
extends List<Nothing> {
elems := [head] ++ tail.elements();
fun elements() : Iterable<E> {return elems;}
}