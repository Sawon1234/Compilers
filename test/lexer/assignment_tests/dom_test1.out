interface Name < Name > { fun name ( ) : Name < Name > ; } class Name ( ) extends Name < Name > { fun name ( ) : Name < Name > { return [ ] ; } } class Name < Name > ( Name name , Name < Name > name ) extends Name < Name > { name := [ name ] ++ name . name ( ) ; fun name ( ) : Name < Name > { return name ; } }